frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -33,15 +34,31 @@ import {
 import { AppResources } from './edit-application-types';
 import { RegistryType } from '../../utils/imagestream-utils';
 import { getHealthChecksData } from '../health-checks/create-health-checks-probe-utils';
-import { detectGitType } from '../import/import-validation-utils';
+import {
+  detectGitType,
+  validationSchema as importValidationSchema,
+} from '../import/import-validation-utils';
 import { getAutoscaleWindow } from '../import/serverless/serverless-utils';
+import { deployValidationSchema } from '../import/deployImage-validation-utils';
+import { validationSchema as jarValidationSchema } from '../import/upload-jar-validation-utils';
 
 export enum CreateApplicationFlow {
   Git = 'Import from Git',
   Dockerfile = 'Import from Dockerfile',
   Container = 'Deploy Image',
+  JarUpload = 'Upload JAR file',
+}
+
+export enum BuildSourceType {
+  Git = 'Git',
+  Binary = 'Binary',
 }
 
+const isFromJarUpload = (type: string): boolean => type === BuildSourceType.Binary;
+
+const getBuildSourceType = (buildConfig: K8sResourceKind): string =>
+  buildConfig?.spec?.source?.type ?? undefined;
```

Comment:
if any value is not there in `buildConfig?.spec?.source?.type` then it'll return `undefined` only so does it make sense to return `undefined` explicitly as well?